tests/components/transmission/test_config_flow.py

```diff
@@ -1,269 +1,158 @@
-    result = await options_flow.async_step_init({CONF_SCAN_INTERVAL: 10})
-    assert result["type"] == data_entry_flow.FlowResultType.CREATE_ENTRY
-    assert result["data"][CONF_SCAN_INTERVAL] == 10
+    assert result2["type"] == FlowResultType.ABORT
+    assert result2["reason"] == "already_configured"
 
 
-async def test_host_already_configured(hass, api):
-    """Test host is already configured."""
+async def test_name_already_configured(hass):
+    """Test name is already configured."""
     entry = MockConfigEntry(
         domain=transmission.DOMAIN,
-        data=MOCK_ENTRY,
-        options={CONF_SCAN_INTERVAL: DEFAULT_SCAN_INTERVAL},
+        data=MOCK_CONFIG_DATA,
+        options={"scan_interval": 120},
     )
     entry.add_to_hass(hass)
 
-    mock_entry_unique_name = MOCK_ENTRY.copy()
-    mock_entry_unique_name[CONF_NAME] = "Transmission 1"
+    mock_entry = MOCK_CONFIG_DATA.copy()
+    mock_entry["host"] = "1.1.1.1"
     result = await hass.config_entries.flow.async_init(
         transmission.DOMAIN,
         context={"source": config_entries.SOURCE_USER},
-        data=mock_entry_unique_name,
-    )
-    assert result["type"] == "abort"
-    assert result["reason"] == "already_configured"
-
-    mock_entry_unique_port = MOCK_ENTRY.copy()
-    mock_entry_unique_port[CONF_PORT] = 9092
-    mock_entry_unique_port[CONF_NAME] = "Transmission 2"
-    result = await hass.config_entries.flow.async_init(
-        transmission.DOMAIN,
-        context={"source": config_entries.SOURCE_USER},
-        data=mock_entry_unique_port,
+        data=mock_entry,
     )
-    assert result["type"] == data_entry_flow.FlowResultType.CREATE_ENTRY
 
-    mock_entry_unique_host = MOCK_ENTRY.copy()
-    mock_entry_unique_host[CONF_HOST] = "192.168.1.101"
-    mock_entry_unique_host[CONF_NAME] = "Transmission 3"
-    result = await hass.config_entries.flow.async_init(
-        transmission.DOMAIN,
-        context={"source": config_entries.SOURCE_USER},
-        data=mock_entry_unique_host,
-    )
-    assert result["type"] == data_entry_flow.FlowResultType.CREATE_ENTRY
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {"name": "name_exists"}
 
 
-async def test_name_already_configured(hass, api):
-    """Test name is already configured."""
+async def test_options(hass: HomeAssistant) -> None:
+    """Test updating options."""
     entry = MockConfigEntry(
         domain=transmission.DOMAIN,
-        data=MOCK_ENTRY,
-        options={CONF_SCAN_INTERVAL: DEFAULT_SCAN_INTERVAL},
+        data=MOCK_CONFIG_DATA,
+        options={"scan_interval": 120},
     )
     entry.add_to_hass(hass)
 
-    mock_entry = MOCK_ENTRY.copy()
-    mock_entry[CONF_HOST] = "0.0.0.0"
-    result = await hass.config_entries.flow.async_init(
-        transmission.DOMAIN,
-        context={"source": config_entries.SOURCE_USER},
-        data=mock_entry,
+    result = await hass.config_entries.options.async_init(entry.entry_id)
```

Comment:
We need to set up the integration with `hass.config_entries.async_setup` before starting the options flow to make sure the config flow is loaded.

Reply:
```suggestion
    with patch(
        "homeassistant.components.transmission.async_setup_entry",
        return_value=True,
    ):
        assert await hass.config_entries.async_setup(entry.entry_id)
        await hass.async_block_till_done()

    result = await hass.config_entries.options.async_init(entry.entry_id)
```